### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by taphu**
See commits for comments.

### CODE REVIEW

File: src/Control/Concurrent/LoadDistribution.hs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  Some notes on the priority queue: The priority portion of the priority queue
+  is a tuple of the form (p :: Int, r :: Int). The first element, @p@, stands
+  for "processes", and indicates the number of simultaneous threads that are
+  using the specific resource. This value is incremented and decremented as
+  threads acquire and release the resource. The second element, @r@, stands for
+  "round robin". Its purpose is to act as a tie breaker between resources that
+  have the same @p@ value by giving priority to whichever resource has been
+  chosen fewer times. It is always incremented and never decremented. Without
+  the @r@ component, the deterministic nature of the priority queue will
+  heavily favor resources that come first in the sorted list of resources.
```
Comment (glasserc) [inline src/Control/Concurrent/LoadDistribution.hs]: This comment probably belongs with the datatype definition.

Comment (taphu) [inline src/Control/Concurrent/LoadDistribution.hs]: You're right.

Comment (taphu) [inline src/Control/Concurrent/LoadDistribution.hs]: I will do that.

Comment (taphu) [inline src/Control/Concurrent/LoadDistribution.hs]: Except that I don't think they belong to the data type, it belongs to the interpretation of the data type, which is embedded in the implementation functions `evenlyDistributed`.

**Comment by taphu** [comment]
Fixed

**Pull request merged**